mosaic-site/src/components/mdx/anchorHeading/AnchorHeading.tsx

```diff
@@ -0,0 +1,79 @@
+import React, {
+  useState,
+  ReactNode,
+  FC,
+  ElementType,
+  PropsWithChildren,
+  HTMLProps,
+} from "react";
+import classnames from "classnames";
+import { Link, LinkProps } from "@jpmorganchase/mosaic-components";
+import styles from "./AnchorHeading.module.css";
+
+interface TypographyProps {
+  children: ReactNode;
+  /** Additional class name for root class override */
+  className?: string;
+  /** Root element type */
+  component?: ElementType;
+  /** Component id */
+  id?: string;
+  /** role */
+  role?: string;
+}
+
+export interface AnchorHeadingProps extends HTMLProps<HTMLHeadingElement> {
+  children: ReactNode[];
+  Component: FC<PropsWithChildren<TypographyProps>>;
+  LinkProps?: LinkProps;
+}
+
+export const AnchorHeading: FC<PropsWithChildren<AnchorHeadingProps>> = ({
+  Component,
+  children,
+  className,
+  id,
+  LinkProps: LinkPropsProp = {},
+  ...rest
+}) => {
+  const [hovered, setHovered] = useState(false);
+
+  const handleMouseEnter = () => {
+    setHovered(true);
+  };
+  const handleMouseLeave = () => {
+    setHovered(false);
+  };
+
+  const { link } = LinkPropsProp;
+  const anchorLink = link || `#${id}`;
+  return (
+    <div className={classnames(className, styles.root)}>
```

Comment:
does this class have to apply to both the div and the Link? also, the classname that comes from props should probably be the last one